### ISSUE TITLE
CloudImageService: Issue with "Host"/"Container" in config

### ISSUE THREAD

**Issue by Telaran** [JimBobSquarePants/ImageProcessor#635]
### Foreword
I've seen several examples where in the security.config the "Host" is provided but not the "Container". So it's "seems" like this should be "the best practice approach"? 

#### The issue/question
Due another issue  #609 I've got the repository (2704e38f062fb3c9267193129eea5cd363887208) and did debug a bit. During that I've encountered the issue where CloudImageService does create a wrong URL and therefore not find the image. If you provide the "container" in the settings it will work as expected.

#### CloudImageService.cs (Line 107):
`Uri uri = new Uri(baseUri, relativeResourceUrl);`

**Values**
_baseUri = "https://xxxx.blob.core.windows.net/pictures/"
relativeResourceUrl = "/1386336459.png"_

Based on Line 107 the **_uri_** will be:
_"https://xxxx.blob.core.windows.net/1386336459.png"_
But should be:
 _"https://xxxx.blob.core.windows.net/pictures/1386336459.png"_

Maybe change to [UriBuilder](http://msdn.microsoft.com/en-us/library/system.uribuilder.aspx) would be a better approach.
```
System.UriBuilder uriBuilder = new System.UriBuilder(baseUri);
uriBuilder.Path += relativeResourceUrl.TrimStart('/');
Uri uri = new Uri(uriBuilder.ToString());
```

### Conclusion
So the question is: What should it be? 
I'm fine with both:  "Host" with the full uri (but then a fix will be needed) or "Host" and "Container" combined.

**Comment by JimBobSquarePants** [comment]
I'm sorry but this makes little sense to me.

The Host property on the CloudImageService expects a host value. 

e.g

`https://xxxx.blob.core.windows.net`

Passing arbitrary path values in addition to that are not supported. Relative paths are not part of the host  of a URL. 

So the correct request URL would be

`/pictures/1386336459.png`

**Comment by Telaran** [comment]
Thanks for your answer.
I've checked again and understand now, that I had a mess with the security.config.
I've had a false assumption on how to use "prefix"-Attribute and that caused me a 404 issue. While I was debugging to get the cause I've ended up to this wrong conclusion.

So everything is fine now. Without prefix-Attribute, without Container-Key and the "Host"-Key without any path.

Sorry for the inconvenience I may have caused.
